R/derive_interp_records.R

```diff
@@ -95,22 +105,32 @@ derive_interp_records <- function(dataset,
 
   # Apply the function within each group and combine the results
   if (is.null(by_vars)) {
-    dataset <- dataset %>%
+    interp_dataset <- dataset %>%
       reframe({
         age <- AGE
-        x <- lapply(select(., all_of(metadata_vars)), fapp, age = age)
+        x <- lapply(select(., all_of(interp_vars)), fapp, age = age)
         as.data.frame(do.call(bind_cols, x))
       }) %>%
       filter(!is.na(AGE))
   } else {
-    dataset <- dataset %>%
+    interp_dataset <- dataset %>%
       group_by(across(!!!syms(map(replace_values_by_names(by_vars), as_label)))) %>%
```

Comment:
```suggestion
      group_by(!!!by_vars, AGEU) %>%
```
Then `AGEU` is populated for all records and the `if` branch above can be removed.

Reply:
`!!!by_vars` is not sufficient. If `by_vars` contains several variables it resolves to an error like "Object not found" : at least `map_chr(replace_values_by_names(by_vars), as_label)` is needed here.